utils/src/main/java/org/nervos/ckb/crypto/secp256k1/ECKeyPair.java

```diff
@@ -44,31 +44,54 @@ public static ECKeyPair createWithKeyPair(KeyPair keyPair) {
   }
 
   public static ECKeyPair createWithPrivateKey(BigInteger privateKey, boolean compressed) {
-    return new ECKeyPair(privateKey, Sign.publicKeyFromPrivate(privateKey, compressed));
+    return new ECKeyPair(privateKey, publicKeyFromPrivate(privateKey, compressed));
   }
 
   public static ECKeyPair createWithPrivateKey(BigInteger privateKey) {
-    return new ECKeyPair(privateKey, Sign.publicKeyFromPrivate(privateKey));
+    return new ECKeyPair(privateKey, publicKeyFromPrivate(privateKey));
   }
 
   public static ECKeyPair createWithPrivateKey(byte[] privateKey) {
     return createWithPrivateKey(Numeric.toBigInt(privateKey));
   }
 
   /**
-   * Sign a hash with the private key of this key pair.
+   * Returns public key from the given private key.
    *
-   * @param transactionHash the hash to sign
-   * @return An {@link ECDSASignature} of the hash
+   * @param privateKey the private key to derive the public key from
+   * @return BigInteger encoded public key
    */
-  public ECDSASignature sign(byte[] transactionHash) {
-    ECDSASigner signer = new ECDSASigner(new HMacDSAKCalculator(new SHA256Digest()));
+  public static BigInteger publicKeyFromPrivate(BigInteger privateKey) {
+    return publicKeyFromPrivate(privateKey, true);
+  }
 
-    ECPrivateKeyParameters privKey = new ECPrivateKeyParameters(privateKey, Sign.CURVE);
-    signer.init(true, privKey);
-    BigInteger[] components = signer.generateSignature(transactionHash);
+  public static BigInteger publicKeyFromPrivate(String privateKeyHex) {
```

Comment:
If this method accept a private key as hex string, why not set the returned public key as hex string too?

Reply:
Yes. We can support many similar methods whose parameters and return results different data types, like BigInteger, hex string and byte[]. Relatively speaking, BigInteger is more generic.

So we can support all combinations, so the question is we need support all combinations ?

Reply:
I means this method accept a string and why not return a string instead of BigInteger ?

Reply:
Fixed. https://github.com/nervosnetwork/ckb-sdk-java/pull/149/commits/0263baf5aad39f61e3a8062fd552c37410001a10